jsapp/js/components/projectDownloads/anonymousExports.es6

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import {bem} from 'js/bem';
+import {actions} from 'js/actions';
+import {
+  EXPORT_STATUSES,
+  EXPORT_REFRESH_TIME,
+  DEFAULT_EXPORT_SETTINGS,
+} from 'js/components/projectDownloads/exportsConstants';
+import {getContextualDefaultExportFormat} from 'js/components/projectDownloads/exportsUtils';
+import exportsStore from 'js/components/projectDownloads/exportsStore';
+import ExportTypeSelector from 'js/components/projectDownloads/exportTypeSelector';
+
+/**
+ * @prop {object} asset
+ */
+export default class AnonymousExports extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      selectedExportType: exportsStore.getExportType(),
+      isPending: false,
+      exportUrl: null,
+    };
+    this.unlisteners = [];
+    this.fetchIntervalId = null;
+    autoBind(this);
+  }
+
+  componentDidMount() {
+    this.unlisteners.push(
+      exportsStore.listen(this.onExportsStoreChange),
+      actions.exports.createExport.completed.listen(this.onCreateExportCompleted),
+      actions.exports.getExport.completed.listen(this.onGetExportCompleted),
+    );
+  }
+
+  componentWillUnmount() {
+    this.unlisteners.forEach((clb) => {clb();});
+  }
+
+  onExportsStoreChange() {
+    this.setState({
+      selectedExportType: exportsStore.getExportType(),
+      exportUrl: null,
+    });
+  }
+
+  onCreateExportCompleted(exportData) {
+    this.fetchExport(exportData.uid);
+  }
+
+  onGetExportCompleted(exportData) {
+    this.prepareFetchInterval(exportData.uid, exportData.status);
+
+    if (exportData.status === EXPORT_STATUSES.complete) {
+      this.setState({
+        isPending: false,
+        exportUrl: exportData.result,
+      });
+      this.downloadUrl(this.state.exportUrl);
+    }
+  }
+
+  onSubmit() {
+    if (this.state.exportUrl) {
+      // we remember the current type download to not make multiple calls
+      this.downloadUrl(this.state.exportUrl);
```

Comment:
You're right that it would be nice, but I didn't want to evercomplicate this. The thing I wanted to achieve is if someone clicks "Export" and the file download native prompt appears and they click "cancel" they don't need to fetch again.

Reply:
I guess I can leave a comment about potential improvement if we see that there is too much usage from the anonymous users